src/main.rs

```diff
@@ -135,52 +183,59 @@ fn main() {
         Default::default()
     };
 
-    if let Some(source) = matches.opt_str("s") {
-        config.source = source;
-    };
+    config.source = matches.value_of("source")
+                           .map(str::to_string)
+                           .unwrap_or(config.source);
 
-    if let Some(dest) = matches.opt_str("d") {
-        config.dest = dest;
-    };
+    config.dest = matches.value_of("destination")
+                         .map(str::to_string)
+                         .unwrap_or(config.dest);
 
-    if let Some(layouts) = matches.opt_str("layouts") {
-        config.layouts = layouts;
-    };
+    config.layouts = matches.value_of("layouts")
+                         .map(str::to_string)
+                         .unwrap_or(config.layouts);
 
-    if let Some(posts) = matches.opt_str("posts") {
-        config.posts = posts;
-    };
+    config.posts = matches.value_of("posts")
+                         .map(str::to_string)
+                         .unwrap_or(config.posts);
 
-    let command = if !matches.free.is_empty() {
-        matches.free[0].clone()
-    } else {
-        print_usage(opts);
-        return;
-    };
+    match matches.subcommand() {
+        ("new", Some(submatches)) => {
+            match submatches.value_of("DIRECTORY") {
+                Some(directory) => config.dest = directory.to_string(),
```

Comment:
It is a bit confusing to use the `config.dest` variable for the destination of the new project. I would create a new variable like in the old code with `getopts`.